idl/forward_request.idl.hh

```diff
@@ -0,0 +1,21 @@
+namespace query {
+struct forward_request {
```

Comment:
"forwarding" as a term refers to when the request is fully processed by the new node, which is not the case here, as the caller does more processing. We'll need to find a better term.

Reply:
delegate?

Reply:
That's more or less the same as forwarding. We're subcontracting some of the work here, but not all of it.

Reply:
Yes,  but the request we forward/delegate also makes sense as a standalone request, and I'd say that it's an implementation detail that we create multiple requests, forward/delegate them and then merge the results. In fact, our first iteration of this project implemented full forwarding - the whole request was simply forwarded from one coordinator to another. Perhaps that's why the name sounds reasonable to me.

Reply:
The nodes we are "forwarding" to: are they expected in turn to read from more nodes (replicas), or only from themselves? In the latter case this is not forwarding, but query processing pushdown to replicas.

Reply:
They are "forwarding" to other coordinator. This node's working title is a "supercoordinator"